packages/block-library/src/image/edit.js

```diff
@@ -73,23 +73,27 @@ class ImageEdit extends Component {
 
 		this.state = {
 			captionFocused: false,
+			hasError: false,
 		};
 	}
 
 	componentDidMount() {
-		const { attributes, setAttributes } = this.props;
+		const { attributes, setAttributes, noticeOperations } = this.props;
 		const { id, url = '' } = attributes;
-
 		if ( ! id && url.indexOf( 'blob:' ) === 0 ) {
 			const file = getBlobByURL( url );
 
 			if ( file ) {
 				mediaUpload( {
 					filesList: [ file ],
+					allowedTypes: ALLOWED_MEDIA_TYPES,
 					onFileChange: ( [ image ] ) => {
 						setAttributes( { ...image } );
 					},
-					allowedTypes: ALLOWED_MEDIA_TYPES,
+					onError: ( message ) => {
+						this.setState( { hasError: true } );
```

Comment:
This seems a really complex problem if we apply a setAttributes here. The attributes are in fact changed in the store, but the components are not rerendered even if caching is disabled on the getBlocks selector. That's why there is a delay on the notice appearing when we use the setAttributes approach.

@noisysocks given that the state approach was also used on the file blocks do you have any insights on why setAttributes does not work for this cases?
I'm fine with the approach being proposed here my only fear is that the fact the setAttributes approach does not work well may be related to some problem that may have an impact in other areas.

Thank you @brentswisher for this PR and for the research you made.

Reply:
It looks like this isn't an issue in master anymore, wondering if #9264 may have fixed this issue as well and my PR is no longer needed?

Reply:
Using `setAttributes()` causes an undo level to be created.